---
title: "Welcome to CoMPaSS Webpage"
sidebar: false
---

The Webpage includes all the code used in paper: "CoMPaSS: A Computational Pipeline for Cross-Platform Concordance Assessment and Navigating Study Design in Microbiome Research"

------------------------------------------------------------------------

Microbiome research has become an integral component of modern biomedical science, offering insights that uniquely complement other omics. To characterize microbial dysbiosis and its impact on human health, researchers have widely employed next-generation sequencing (NGS) technologies, including 16S rRNA gene sequencing and shotgun metagenomic sequencing. However, these approaches are susceptible to technological artifacts that may introduce method-specific bias into analyses and undermine the replicability of discoveries. Researchers often struggle to select appropriate sequencing methods during the study design stage, as they strive to optimize the budget, taxonomic resolution, and analytical accuracy. While 16S rRNA sequencing is a cost-effective and reliable method that offers acceptable taxonomic resolution, shotgun metagenomic sequencing provides higher-resolution profiles and functional insights, albeit with increased cost. To address this challenge, we introduce CoMPaSS (Concordance of Microbiome Sequencing Platforms and Study Initiation Strategy), a computational pipeline designed to enable precise and head-to-head comparison of six downstream analytical tools in efficacy and to navigate microbiome study design. CoMPaSS offers a framework for evaluating cross-platform concordance in statistical inference across multiple contexts, from community diversity indices to taxonomic profiling, and offers integrated tools for power analysis, sample size estimation, and cost assessment to support comprehensive study planning. Based on an extensive simulation study of in-silico microbiota and three real-world microbiome studies in population-based cohorts (i.e., ORIGIN, IBD, TEDDY), CoMPaSS reveals desirable concordance between platforms at higher taxonomic levels, but significant discrepancies at finer taxonomic resolutions and among rare taxaâ€”highlighting the importance of sequencing platform selection in determining study outcomes.

::: callout-note
## This webpage include

- R code for simulation analysis

- R code for real data analysis

:::


